streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroStreamsSnapshotDeserializer.java

```diff
@@ -65,7 +65,10 @@ private Object convertValue(Object deserializedObj) {
             List<Schema.Field> fields = indexedRecord.getSchema().getFields();
             ImmutableMap.Builder<String, Object> keyValues = ImmutableMap.builder();
             for (Schema.Field field : fields) {
-                keyValues.put(field.name(), convertValue(indexedRecord.get(field.pos())));
+                Object currentValue = convertValue(indexedRecord.get(field.pos()));
+                if (currentValue != null) {
```

Comment:
@satishd I understand that a key containing null value is not equivalent to not having key at all. But how does that matter here? Is it going to break anything? Can you give an example? We need this map to be immutable since its going to be sent across in storm. I am still not clear on exact solution you are suggesting? Is there an immutable map that allows null values?